atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -293,12 +297,14 @@ static LockAndTimestampServices createRawServices(
             Environment env,
             Supplier<RemoteLockService> lock,
             Supplier<TimestampService> time,
+            Supplier<TimestampStoreInvalidator> invalidator,
```

Comment:
I don't see much reason for this being a supplier, the other services are suppliers because the AwaitingLeadershipProxy requires suppliers for creating the services when it's elected leader.  With this in mind we may want to memoize the supplier as well though doesn't seem as improtant.